lib/src/generator/templates.dart

```diff
@@ -35,15 +38,26 @@ ${nonnullOfFunction ? """
     final localeName = Intl.canonicalizedLocale(name);${otaEnabled ? '\n${_generateMetadataSetter()}' : ''} 
     return initializeMessages(localeName).then((_) {
       Intl.defaultLocale = localeName;
-      $className.${nonnullOfFunction ? '_internalCurrent' : 'current'} = $className();
+      final delegate = $className();
+      $className.${nullableDelegate ? 'current' : '_current'} = delegate;
  
-      return $className.current${nonnullOfFunction ? '' : '!'};
+      return delegate;
     });
   } 
 
-  static $className${nonnullOfFunction ? '' : '?'} of(BuildContext context) {
-    return Localizations.of<$className>(context, $className)${nonnullOfFunction ? '!' : ''};
+${nullableDelegate ? """
+  static $className? of(BuildContext context) {
```

Comment:
Since we now have `maybeOf(context)` as well, the `of(context)` variant can directly be included? Then we can completely remove the `nullDelegate` option. 

Thus it becomes the following:
```
  static $className of(BuildContext context) {
    final delegate = $className.maybeOf(context);
    assert(delegate != null, 'No instance of $className present in the widget tree. Did you add $className.delegate in localizationsDelegates?');
    return delegate!;
  }

  static $className? maybeOf(BuildContext context) {
    return Localizations.of<$className>(context, $className);
  }
```

This eliminates the need for `${nullableDelegate ? <codeIfNullable> : <code if not nullable>}`.